at_onboarding_cli/lib/src/at_onboarding_service_impl.dart

```diff
@@ -0,0 +1,290 @@
+        atKeysMap[AuthKeyType.selfEncryptionKey]);
+    atKeysMap[AuthKeyType.encryptionPublicKey] = EncryptionUtil.encryptValue(
+        atKeysMap[AuthKeyType.encryptionPublicKey],
+        atKeysMap[AuthKeyType.selfEncryptionKey]);
+    atKeysMap[AuthKeyType.encryptionPrivateKey] = EncryptionUtil.encryptValue(
+        atKeysMap[AuthKeyType.encryptionPrivateKey],
+        atKeysMap[AuthKeyType.selfEncryptionKey]);
+    //generating .atKeys file at path provided in onboardingConfig
+    String filePath = path.join(
+        atOnboardingPreference.downloadPath!, '${_atSign}_key.atKeys');
+    IOSink atKeysFile = File(filePath).openWrite();
+    atKeysFile.write(jsonEncode(atKeysMap));
+    await atKeysFile.flush();
+    await atKeysFile.close();
+    logger.finer('atKeys file saved at ${atOnboardingPreference.downloadPath}');
+    atOnboardingPreference.atKeysFilePath = filePath;
+  }
+
+  ///back-up encryption keys to local secondary
+  Future<void> _persistKeysLocalSecondary(Map? _atKeysMap, bool isPkam) async {
+    //get decrypt atKeys file data
+    if (isPkam) {
+      _atKeysMap = await _decryptAtKeysFile(
+          await _readAtKeysFile(atOnboardingPreference.atKeysFilePath));
+    }
+    //backup keys into local secondary
+    if (_atKeysMap != null) {
+      bool? response = await _atClient
+          ?.getLocalSecondary()
+          ?.putValue(AT_PKAM_PUBLIC_KEY, _atKeysMap[AuthKeyType.pkamPublicKey]);
+      logger.finer('pkamPublicKey persist status $response');
+      response = await _atClient?.getLocalSecondary()?.putValue(
+          AT_PKAM_PRIVATE_KEY, _atKeysMap[AuthKeyType.pkamPrivateKey]);
+      logger.finer('pkamPrivateKey persist status $response');
+      response = await _atClient?.getLocalSecondary()?.putValue(
+          AT_ENCRYPTION_PUBLIC_KEY,
+          _atKeysMap[AuthKeyType.encryptionPublicKey]);
+      logger.finer('encryptionPublicKey persist status $response');
+      response = await _atClient?.getLocalSecondary()?.putValue(
+          AT_ENCRYPTION_PRIVATE_KEY,
+          _atKeysMap[AuthKeyType.encryptionPrivateKey]);
+      logger.finer('encryptionPrivateKey persist status $response');
+      response = await _atClient?.getLocalSecondary()?.putValue(
+          AT_ENCRYPTION_SELF_KEY, _atKeysMap[AuthKeyType.selfEncryptionKey]);
+      logger.finer('self encryption key persist status $response');
+    } else {
+      logger.severe('atKeysMap is null');
+    }
+  }
+
+  @override
+  Future<bool> authenticate() async {
+    atOnboardingPreference.privateKey ??= _getPkamPrivateKey(
+        await _readAtKeysFile(atOnboardingPreference.atKeysFilePath));
+    if (atOnboardingPreference.privateKey == null) {
+      throw UnAuthenticatedException(
+          'either of private key or .atKeys file not provided');
+    } else {
+      _atClient ??= await getAtClient();
+      _isPkamAuthenticated =
+          (await _atLookup?.authenticate(atOnboardingPreference.privateKey))!;
```

Comment:
As you strictly wating for true/false value, you can actualy use `_atLookup!` instead of `_atLookup?`. Incase if you think `_atLookup` will be null, then write a if condition for it. The thing what happens is, If `_atLookup` is null, then `_isPkamAuthenticated` will be null. The above mentioned is a possible solution or there is an alternative solution, here it is :

```dart
_isPkamAuthenticated = await _atLookup?.authenticate(atOnboardingPreference.privateKey) ?? false;
```